core/Pos/Core/Context.hs

```diff
@@ -9,9 +24,57 @@ module Pos.Core.Context
 
 import           Universum
 
-import           Pos.Core.Address (addressHash, makePubKeyAddress)
-import           Pos.Core.Types   (Address, StakeholderId)
-import           Pos.Crypto       (PublicKey, SecretKey, toPublic)
+import           Control.Lens       (makeLenses)
+import           Data.Reflection    (Given (..), give)
+
+import           Pos.Core.Address   (addressHash, makePubKeyAddress)
+import           Pos.Core.Types     (Address, BlockCount (..), StakeholderId, SlotCount)
+import           Pos.Crypto         (PublicKey, SecretKey, toPublic)
+import           Pos.Core.Constants (staticBlkSecurityParam)
+
+-- | Core constants. They should be really constant and never change.
+data CoreConstants = CoreConstants
+    { _ccBlkSecurityParam :: !BlockCount
+    }
+
+makeLenses ''CoreConstants
+
+type HasCoreConstants = Given CoreConstants
+
+giveConsts :: BlockCount -> (HasCoreConstants => r) -> r
```

Comment:
wtf? you can consider this function as `give . CoreConstants`. constructor and this function are absolutely same.
if you meant 
```
field1 <- fromIntegral a
field2 <- fromIntegral b
...
```
you can also confuse records